nixos/modules/services/torrent/torrentstream.nix

```diff
@@ -0,0 +1,72 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.torrentstream;
+  dataDir = "/var/lib/torrentstream/";
+in
+{
+  options.services.torrentstream = {
+    enable = lib.mkEnableOption (lib.mdDoc "TorrentStream daemon");
+    package = lib.mkOption {
+      type = lib.types.package;
+      example = lib.literalExpression "pkgs.torrentstream";
+      description = lib.mdDoc ''
+        TorrentStream package to use.
+      '';
+    };
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 5082;
+      description = lib.mdDoc ''
+        TorrentStream port.
+      '';
+    };
+    openFirewall = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Open ports in the firewall for TorrentStream daemon.
+      '';
+    };
+    address = lib.mkOption {
+      type = lib.types.str;
+      default = "127.0.0.1";
+      description = lib.mdDoc ''
+        Address to listen on.
+      '';
+    };
+  };
+  config = lib.mkIf cfg.enable {
+    services.torrentstream.package = lib.mkDefault pkgs.torrentstream;
+    users.groups.torrentstream = { };
+    users.users.torrentstream = {
+      isSystemUser = true;
+      group = "torrentstream";
+      description = "TorrentStream Daemon user";
+      home = dataDir;
+    };
```

Comment:
Just to make sure, have you researched the possibility of making this `DynamicUser`? If it's possible, I think it's often a better alternative.

Reply:
I'm not gonna advocate for `DynamicUser` if it causes a lot of trouble, but maybe it would be better to use `StateDirectory` as a default `dataDir`, and just assume people actually create the directory in question if they override it?